### ISSUE TITLE
Properties from inherited interfaces are not serialized

### ISSUE THREAD

**Issue by askmoller** [Blazored/LocalStorage#119]
Hosting Model: Blazor Webassembly
Framework: .NET 5
Browser: Google Chrome - version 88.0.4324.104
IDE: Visual Studio 2019 Preview - version 16.9.0 Preview 2.0

A simplified example:

```
public interface ITransportObject
{
    double Speed { get; set; }
}
public interface ICar : ITransportObject
{
    string ModelName { get; set; }
}
```

In the above structure only the `ModelName` property of `ICar` will be serialized into the localstorage when calling:
`await LocalStorage.SetItemAsync(nameof(PropertyOfTypeICar), PropertyOfTypeICar);`

The `Speed` property of `ITransportObject` will not be serialized. Can this be fixed or do I need to do something special?

**Comment by askmoller** [comment]
I have made a simple example that illustrates the issue above:
[BlazorLocalStorage.zip](https://github.com/Blazored/LocalStorage/files/5920947/BlazorLocalStorage.zip)

Its made from the template in visual studio. I have added the file IndexBase.cs, a button to Index.razor and BlazoredLocalStorage to the service in Program.cs.

**Comment by askmoller** [comment]
I looked a bit more into this and the issue is from the JsonSerializer. More info here: https://github.com/dotnet/runtime/issues/31742

TLDR: "If you want "deeply polymorphic" you need to serialize as object. Any other type, including getting the type by generic inference, is intentionally static typed, not polymorphic."

In the code example from BlazorLocalStorage.zip, the ICar property can be save by
`await LocalStorage.SetItemAsync<object>(nameof(Car), Car);`
